File: TaskBoard.BLL/Models/ViewModels/CardListVm.cs
using AutoMapper;
using TaskBoard.BLL.Common.Mapping;
using TaskBoard.DAL.Data.Entities;

namespace TaskBoard.BLL.Models.ViewModels.List;

public class CardListVm : IMapWith<CardList>
{
    public Guid Id { get; set; }
    public string Name { get; set; }
    public IReadOnlyList<CardVmList> Cards { get; set; }
    
    public void Mapping(Profile profile)
    {
        profile.CreateMap<CardList, CardListVm>();
    }
}

File: TaskBoard.UI/src/app/models/view-models/cardlist-vm.ts
export class CardlistVm{
  id= '';
  name= '';
  cards: CardVmList[] = [];
}

export class CardVmList{
  id= '';
  name= '';
  description= '';
  dueDate: Date = new Date();
  priorityName= '';
  cardListId='';
}
